readme.md

```diff
@@ -1 +1,59 @@
-# gravitational
+# Scheduler-Worker Cluster Architecture
+
+
+# Goal
+
+*   Build a scheduler and worker architecture that supports GRPC communication.
+
+# General Overview
+
+*   Support single scheduler and multiple workers.
+*   The scheduler must be aware of all workers' states. (requires the workers to register and deregister when going offline)
+*   The scheduler and the workers authenticate with each other via SSL.
+*   The scheduler and workers, both need to have GRPC-APIs. 
+    *   The scheduler GRPC-API support:
+        *   Register a worker
+        *   Deregister a worker
+    *   The worker GRPC-API support:
+        *   Start a job
+        *   Stop a job
+        *   Return the status of a job
+        *   Return a stream of output for a running job.
+
+
+# Scheduler Overview
+
+* [Scheduler Details](scheduler/scheduler.md)
+* All config parameters are specified in the config.toml file
+* When a worker registers, a UUID is assigned to the worker and worker details are kept in a map.
```

Comment:
You are using in-memory model, which is fine. Please write down (but don't implement) some notes on what backend do you want to pick for persistence and why.